lib/Twig/Loader/Filesystem.php

```diff
@@ -302,4 +302,34 @@ private function isAbsolutePath($file)
             || null !== parse_url($file, PHP_URL_SCHEME)
         ;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getTemplateFreshness(Twig_CacheInterface $cache)
+    {
+        $cacheKey = $cache->generateKey(
+            md5(serialize($this->cache)),
```

Comment:
this cache key will depend of which templates have already been loaded in this process, and of the order in which they have been loaded. This looks like a bad idea to me.